docs/rfc/rfc-2.adoc

```diff
@@ -0,0 +1,811 @@
+  public key by compressing the public key; this is briefly covered in the
+  implementation <<output-script>> section.
+- One Bitcoin address per signer, specified on the client in a way detailed
+  in the section on <<recovery-address>>.
+- The keep's signature on the sighash produced by the previous items,
+  computed via the threshold ECDSA protocol in a way detailed in the section on
+  the <<signing-protocol>>.
+- F, the desired fee, which is a function of the going feerate on the Bitcoin
+  network and the number of bytes in the final transaction.
+
+The desired fee is the most problematic one to procure under the goal of not
+strictly requiring a Bitcoin node connection, as generally a Bitcoin node
+provides estimates of the feerates that will achieve transaction confirmation
+within a certain amount of time. A short-term solution is proposed in the
+section on <<bitcoin-fee>>.
+
+[[recovery-address]]
+===== Specifying an Operator's Bitcoin Recovery Address
+
+To specify a given operator's Bitcoin Recovery Address, this proposal
+suggests adding a single key to the `Extensions.TBTC.BTCRefunds` config
+entry, `BeneficiaryAddress`. The value for this key should be one of:
+
+- A simple BTC address.
+- An xpub, ypub, or zpub that allows the client to derive new addresses for
+  each refund.
+
+The challenge with a *pub is that, to avoid reuse, it needs to do one of two
+things:
+
+- Be able to check, on-chain, whether a given address has been used; or,
+- Have a *pub that is only meant for use by the client, and have the client
+  track the latest address generated (as an index starting at 0).
+
+To avoid _strictly requiring_ a connection to a Bitcoin node, this RFC
+proposes tracking the latest address generated, and allowing an additional
+config key, `TargetDerivationStartIndex`, that sets a starting point for such
+derivations. The client should write the updated index to the `current/`
+directory that tracks active key shares, to an encrypted file named
+`current-btc-refund-derivation-index`. The operator should then seek to
+obtain from the staker an extended public key that is only used for this
+purpose. In cases where in-memory/on-disk derivation index and config file index
+both exist, the highest index should be used.
+
+If an optional Bitcoin node connection is available (see <<electrs>>), the
+client should additionally check a derived address for a balance before
+publishing. To do this, a GET request to `/address/<address>` should be issued,
+and the JSON response inspected for a `tx_count` of 0. When such a connection is
+available, if the client finds an address is already used, it should increment
+the derivation index and check again. If the GET request fails, the index should
+be assumed to be valid, as if the Bitcoin node connection were not available.
+
+==== Modification of key material archiving
+
+In the current client, key material is archived as soon as the ECDSA keep
```

Comment:
what is archiving key material?

Reply:
The client keeps key material in two places: the set of `current` keys and the set of `archive` keys. These are represented as directories on the filesystem. At startup, the client loads up the `current` keys, while `archive` keys are ignored. Archiving key material basically means moving keys from `current` to `archive`.

Reply:
Adding some code references...

Checking keep status on client start and removing keep from the registry if needed:
https://github.com/keep-network/keep-ecdsa/blob/f240e81002a7b26eb6326be068dbf80b53bc0a18/pkg/client/client.go#L123-L127

On a (confirmed) keep closed event:
https://github.com/keep-network/keep-ecdsa/blob/f240e81002a7b26eb6326be068dbf80b53bc0a18/pkg/client/client.go#L829-L830

On a (confirmed) keep terminated event:
https://github.com/keep-network/keep-ecdsa/blob/f240e81002a7b26eb6326be068dbf80b53bc0a18/pkg/client/client.go#L908-L909

It goes down to the keep registry:
https://github.com/keep-network/keep-ecdsa/blob/f240e81002a7b26eb6326be068dbf80b53bc0a18/pkg/registry/keeps.go#L69-L80

And finally, to the on-disk storage implementation:
https://github.com/keep-network/keep-common/blob/34905d2fe019015fb16dd6edc37da16efa988b87/pkg/persistence/disk_persistence.go#L136-L149